packages/analytics/src/client/client.ts

```diff
@@ -0,0 +1,121 @@
+import { AnalyticsClientConfig, GenericAnalyticsClient, JsonMap } from '../utils/types.shared';
+
+export type AnalyticsClientOptions = Pick<
+  AnalyticsClientConfig,
+  'defaultProperties' | 'defaultTraits'
+>;
+
+export class AnalyticsClient<T extends GenericAnalyticsClient> {
```

Comment:
Why the choice here to use a class over a function? We just refactored our api clients away from using classes.

Reply:
I would like to store a reference to `options` and creating a class seemed to be the best way. WHat is the disadvantage of a class over a function?

Reply:
Nothing inherently wrong using a `class` here, and think it's okay for a package like this, but I think Fara's mentioning because we typically favour a more functional approach. Classes often lead to inheritance over composition, mutative methods over pure functions etc. Use of classes in Javascript is a hotly debated topic.

Reply:
Yep, was just questioning based on Kyran's feedback here. Some suggest to avoid classes in JS if possible, so was just curious about your choice.

Reply:
I have historically written in codebases that prefer one or the other. No strong preference but depending on the interface I like classes more. Happy to conform my code to our preferences.